app/main/helpers/validation.py

```diff
@@ -0,0 +1,185 @@
+from functools import reduce
+from operator import add
+import re
+import six
+from werkzeug.datastructures import ImmutableOrderedMultiDict
+
+EMAIL_REGEX = r'^[^@^\s]+@[^@^\.^\s]+(\.[^@^\.^\s]+)+$'
+
+VALIDATORS = {}
```

Comment:
@allait: We could. The only thing I'm concerned about with that is that the registration of the slug-to-class mapping would not be done directly next to the class. If we had, say, 5 or 10 frameworks, it wouldn't be intuitively obvious where the mapping is done.